app/models/chargeback_vm.rb

```diff
@@ -158,6 +158,11 @@ def self.report_col_options
 
   def get_rate_parents(perf)
     @enterprise ||= MiqEnterprise.my_enterprise
-    [perf.parent_host, perf.parent_ems_cluster, perf.parent_storage, perf.parent_ems, @enterprise, perf.resource.try(:tenant)]
+    [perf.parent_host || perf.resource.host,
+     perf.parent_ems_cluster || perf.resource.ems_cluster,
+     perf.parent_storage || perf.resource.storage,
+     perf.parent_ems || perf.resource.ext_management_system,
+     @enterprise,
+     perf.resource.try(:tenant)]
```

Comment:
Notes
 * This list is later passed to the method that finds applicable rate/s (in `get_rates`).
 * We cache rates (probably for performance reasons). 
 * We need to update caching mechanism, when amending rate selection. -> please add `resource_id` into `hash_features_affecting_rate`

Another idea:
 * Given #11648 :yellow_heart:  has beautiful performance boost :fireworks: (less top-level cycles), perhaps we don't need the caching anymore. :question: dunno. :bulb: